src/components/graph/menus/network-modifications/DescriptionRenderer.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { SetStateAction, useCallback, useState } from 'react';
+import { Tooltip } from '@mui/material';
+import { useIsAnyNodeBuilding } from '../../../utils/is-any-node-building-hook';
+import { useSelector } from 'react-redux';
+import { AppState } from '../../../../redux/reducer';
+import IconButton from '@mui/material/IconButton';
+import type { UUID } from 'node:crypto';
+import { setModificationDescription } from '../../../../services/study/network-modifications';
+
+export interface DescriptionRendererProps extends ICellRendererParams<NetworkModificationMetadata> {
+    setModifications: React.Dispatch<SetStateAction<NetworkModificationMetadata[]>>;
+    hoveredRowIndex: number;
+}
+
+const DescriptionRenderer = (props: DescriptionRendererProps) => {
+    const { data, api, node } = props;
+    const studyUuid = useSelector((state: AppState) => state.studyUuid);
+    const currentNode = useSelector((state: AppState) => state.currentTreeNode);
+    const [isLoading, setIsLoading] = useState(false);
+    const isAnyNodeBuilding = useIsAnyNodeBuilding();
+    const mapDataLoading = useSelector((state: AppState) => state.mapDataLoading);
+    const [openDescModificationDialog, setOpenDescModificationDialog] = useState(false);
+
+    const modificationUuid = data?.uuid;
+    const description: string | undefined = data?.description;
+    const empty: boolean = !description;
+
+    const updateModification = useCallback(
+        async (uuid: UUID, descriptionRecord: Record<string, string>) => {
+            setIsLoading(true);
+
+            return setModificationDescription(studyUuid, currentNode?.id, uuid, descriptionRecord.description).finally(
+                () => {
+                    setIsLoading(false);
+                }
+            );
+        },
+        [studyUuid, currentNode?.id]
+    );
+
+    const handleDescDialogClose = useCallback(() => {
+        setOpenDescModificationDialog(false);
+        api.stopEditing();
+    }, [api]);
+
+    const handleModifyDescription = () => {
+        if (isLoading) {
+            return;
+        }
+        setOpenDescModificationDialog(true);
+    };
+
+    return (
+        <>
+            {openDescModificationDialog && modificationUuid && (
+                <DescriptionModificationDialog
+                    open
+                    description={description ?? ''}
+                    elementUuid={modificationUuid}
+                    onClose={handleDescDialogClose}
+                    updateElement={updateModification}
+                />
+            )}
+            <Tooltip title={description} arrow>
+                <IconButton
+                    color="primary"
+                    onClick={handleModifyDescription}
+                    disabled={isLoading || isAnyNodeBuilding || mapDataLoading}
+                >
+                    <EditNoteIcon empty={empty} hidden={node.rowIndex !== props.hoveredRowIndex} />
```

Comment:
I don't mind not destructuring props, but here it's not coherent with the other props.